base/range.jl

```diff
@@ -1038,3 +1038,27 @@ function +(r1::StepRangeLen{T,S}, r2::StepRangeLen{T,S}) where {T,S}
 end
 
 -(r1::StepRangeLen, r2::StepRangeLen) = +(r1, -r2)
+
+# Modular arithmetic on ranges
+
+"""
+    mod(x::Integer, range)
+
+Find `y` in `range` such that `x` ≡ `y` (mod `n`), where `n = length(range)`.
+The range must contain only integers, and have step +1 or -1.
+
+See also: [`mod1`](@ref).
+
+# Examples
+```jldoctest
+julia> mod(0, Base.OneTo(3))
+3
+
+julia> mod(3, 0:2)
+0
+```
+"""
+mod(i::Integer, r::OneTo) = mod1(i, last(r))
+mod(i::Integer, r::AbstractUnitRange{<:Integer}) = mod(i-first(r), length(r)) + first(r)
+mod(i::Integer, r::StepRange) = abs(r.step)==1 ? mod(i-minimum(r), length(r)) + minimum(r) :
```

Comment:
I'm not sure about supporting -1 steps. It kinda feels like it should wrap around the other way.  That is, like this?

![tmp](https://user-images.githubusercontent.com/154641/61749362-976dab00-ad70-11e9-98e9-36b70a78304c.png)

I also think it would be totally reasonable to only support `AbstractUnitRange`s — we limit many index-like computations to those particular subtypes.

Reply:
OK, I agree there's no real need, let's just simplify to step +1.